clippy_lints/src/utils/hir_utils.rs

```diff
@@ -448,8 +448,7 @@ impl<'a, 'tcx: 'a> SpanlessHash<'a, 'tcx> {
                     CaptureClause::CaptureByValue => 0,
                     CaptureClause::CaptureByRef => 1,
                 }.hash(&mut self.s);
-                let value = &self.cx.tcx.hir.body(eid).value;
-                self.hash_expr(value);
```

Comment:
Hmm, I wonder if this is due to my changes in https://github.com/rust-lang/rust/pull/53177 -- specifically the redundant borrow optimization. It doesn't seem *wrong* to me that these fixes are working but it *is* making me nervous. I wonder what changed here.

Reply:
people who are curious to learn more here may want to read my notes starting here: https://github.com/rust-lang/rust/issues/52934#issuecomment-418144787